### PR TITLE
Feature/nft subgraph

### PR DESCRIPTION
**Pull request by novaknole**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: [ID]()

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/subgraph/schema.graphql
Comment (Michael-A-Heuer) [inline packages/subgraph/schema.graphql:39]: Better use `NativeBalance` as we will also be on Polygon and other chains/layers where the native token is not ETH

File: packages/subgraph/schema.graphql
Comment (Michael-A-Heuer) [inline packages/subgraph/schema.graphql:98]: `NativeTransfer`

File: packages/subgraph/schema.graphql
Comment (novaknole) [reply packages/subgraph/schema.graphql]: done https://github.com/aragon/core/pull/251/commits/72d501bfd5d066a003a534043150f5f6667b75b4

File: packages/subgraph/schema.graphql
Comment (novaknole) [reply packages/subgraph/schema.graphql]: https://github.com/aragon/core/pull/251/commits/72d501bfd5d066a003a534043150f5f6667b75b4

File: packages/subgraph/schema.graphql
Comment (brickpop) [inline packages/subgraph/schema.graphql:160]: TO DO

File: packages/subgraph/src/packages/admin/admin.ts
Comment (brickpop) [inline packages/subgraph/src/packages/admin/admin.ts:1]: Used?

File: packages/subgraph/src/utils/tokens/erc20.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/erc20.ts:148]: Wouldn't this mean the DAO has sent money on behalf of someone else that approved the DAO as a spender?

Not sure if we should still store it as the DAO acted / did something

File: packages/subgraph/src/utils/tokens/erc721.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/erc721.ts:178]: Same as before

File: packages/subgraph/src/utils/tokens/erc721.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/erc721.ts:204]: Same as before

File: packages/subgraph/src/utils/tokens/erc20.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/erc20.ts:121]: Why is this ambiguous? 

`from == to` makes no sense ofc, but I am not sure if we shouldn't handle these edge cases differently than simply returning.

Isn't this, formally, still a tx / a transfer that we should index?

File: packages/subgraph/src/utils/tokens/eth.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/src/utils/tokens/eth.ts:69]: Same as before

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:80]: Do we have a follow-up task to add these tests?

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:130]: Do we have a follow-up task to add these tests?

**Review by Michael-A-Heuer — COMMENTED** [review]
I have some uncertainty about how we handle edge cases such as:

- the DAO being the spender of a transaction with external `from` and `to` address for (ERC20, ERC721)
- the DAO sending tokens (Native, ERC20, ERC721) to itself.

If this was a proposal that went through and got executed, don't we want to still be able to show these txns in the finance section? 
For example, there could be an admin who accidentally sent tokens with `from == to`. If it shows up in the finance sections, this would allow him to notice.

File: packages/subgraph/src/packages/admin/admin.ts
Comment (novaknole) [reply packages/subgraph/src/packages/admin/admin.ts:1]: done

File: packages/subgraph/schema.graphql
Comment (novaknole) [reply packages/subgraph/schema.graphql:160]: removed

Comment (novaknole) [reply packages/subgraph/schema.graphql:160]: removed todo, but will look into it later on the next PR.

File: packages/subgraph/src/dao/dao.ts
Comment (novaknole) [reply packages/subgraph/src/dao/dao.ts:121]: removed. might decide to do it in future pr.

File: packages/subgraph/tests/dao/dao.test.ts
Comment (novaknole) [reply packages/subgraph/tests/dao/dao.test.ts:80]: Yes, starting to work on this now..

File: packages/subgraph/tests/dao/dao.test.ts
Comment (novaknole) [reply packages/subgraph/tests/dao/dao.test.ts:130]: yep. working on it.

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:308]:
```suggestion
    // Check NativeTransfer
```

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:524]:
```suggestion
        // Check ERC20Transfer
```

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:604]:
```suggestion
        // Check ERC20Transfer
```

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:527]: What needs to be fixed here?

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:590]:
```suggestion
        // check ERC20Contract entity
```

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:607]: What needs to be fixed here?

Comment (novaknole) [reply packages/subgraph/tests/dao/dao.test.ts:607]: pushed. no worries.

File: packages/subgraph/schema.graphql
Comment (novaknole) [reply packages/subgraph/schema.graphql]: yep. done.

File: packages/subgraph/tests/dao/dao.test.ts
Comment (Michael-A-Heuer) [inline packages/subgraph/tests/dao/dao.test.ts:500]: I would have called this amount. However, I can approve

**Review by Michael-A-Heuer — DISMISSED** [review]
Work will be continued, so good to go to unblock others

File: packages/subgraph/scripts/deploy-subgraph.sh
Comment (mathewmeconry) [reply packages/subgraph/scripts/deploy-subgraph.sh:48]: just use a  `.env`

File: packages/subgraph/scripts/deploy-subgraph.sh
Comment (mathewmeconry) [inline packages/subgraph/scripts/deploy-subgraph.sh:49]: revert that back to satsuma before merge

File: packages/subgraph/src/utils/tokens/erc20.ts
Comment (mathewmeconry) [inline packages/subgraph/src/utils/tokens/erc20.ts:45]: same here. skip checks if an entity for this token is already present

File: packages/subgraph/src/utils/tokens/erc721.ts
Comment (mathewmeconry) [inline packages/subgraph/src/utils/tokens/erc721.ts:93]: do the TODO

**Review by mathewmeconry — CHANGES_REQUESTED** [review]

File: packages/subgraph/schema.graphql
Comment (mathewmeconry) [reply packages/subgraph/schema.graphql:57]: because they don't change our balance and thus don't touch the treasury

File: packages/subgraph/schema.graphql
Comment (novaknole) [reply packages/subgraph/schema.graphql:51]: 100%. I know(I got this on my roadmap, but this PR took my energy and amazing tests, so have to give it up for a while to stay sane..

File: packages/subgraph/schema.graphql
Comment (novaknole) [reply packages/subgraph/schema.graphql:57]: The alignment was `yes`. So really don't care !

File: packages/subgraph/src/utils/tokens/erc20.ts
Comment (mathewmeconry) [reply packages/subgraph/src/utils/tokens/erc20.ts:15]: dumb me. you are doing that already

File: packages/subgraph/src/utils/tokens/erc721.ts
Comment (novaknole) [reply packages/subgraph/src/utils/tokens/erc721.ts]: There was a comment remaining, it's actually done. So I removed the TODO comment.

File: packages/subgraph/scripts/deploy-subgraph.sh
Comment (novaknole) [reply packages/subgraph/scripts/deploy-subgraph.sh]: done

File: packages/subgraph/scripts/deploy-subgraph.sh
Comment (novaknole) [reply packages/subgraph/scripts/deploy-subgraph.sh]: done

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**
